src/qt/walletcontroller.cpp

```diff
@@ -342,3 +342,21 @@ void OpenWalletActivity::open(const std::string& path)
         QTimer::singleShot(0, this, &OpenWalletActivity::finish);
     });
 }
+
+LoadWalletsActivity::LoadWalletsActivity(WalletController* wallet_controller, QWidget* parent_widget)
+    : WalletControllerActivity(wallet_controller, parent_widget)
+{
+}
+
+void LoadWalletsActivity::load()
+{
+    showProgressDialog(tr("Loading wallets…"));
+
+    QTimer::singleShot(0, worker(), [this] {
+        for (std::unique_ptr<interfaces::Wallet>& wallet : node().walletClient().getWallets()) {
+            m_wallet_controller->getOrCreateWallet(std::move(wallet));
+        }
```

Comment:
In commit "qt, wallet: Add LoadWalletsActivity" (6edff4dfa508100564c3152bbccc7b37ac777528)

Note: This is not new code, this is existing code which is just being moved. The original code is deleted in the next commit. I would squash two commits for more clarity here, but it's not important, and of course the best way to split and group commits is subjective.